cartridges/int_adyen_overlay/cartridge/scripts/util/AdyenHelper.ds

```diff
@@ -313,67 +313,122 @@ var __AdyenHelper: Object = {
+            }
+            if (order.customerEmail) {
+                shopperObject['shopperEmail'] = order.customerEmail;
+            }
+            if (empty(order.customerEmail) && profile && profile.getEmail()) {
+                shopperObject['shopperEmail'] = profile.getEmail();
             }
-        }
 
-        var shopperIP = !empty(request) && !empty(request.getHttpRemoteAddress()) ? request.getHttpRemoteAddress() : '';
+            var shopperDetails = {
+                "firstName": order.getBillingAddress().firstName,
+                "gender": gender,
+                "infix": "",
+                "lastName": order.getBillingAddress().lastName
+            };
 
-        if (!empty(shopperIP)) {
-            shopperObject['shopperIP'] = shopperIP;
+            shopperObject['shopperName'] = shopperDetails;
         }
 
-        if (!empty(customerEmail)) {
-            shopperObject['shopperEmail'] = customerEmail;
+        if (profile && profile.getCustomerNo()) {
+            shopperObject['shopperReference'] = profile.getCustomerNo();
+        } else {
+            if (order && order.getCustomerNo()) {
+                shopperObject['shopperReference'] = order.getCustomerNo();
+            }
         }
 
-        if (!empty(customerID)) {
-            shopperObject['shopperReference'] = customerID;
+        var shopperIP = !empty(request) && !empty(request.getHttpRemoteAddress()) ? request.getHttpRemoteAddress() : null;
+        if (shopperIP) {
+            shopperObject['shopperIP'] = shopperIP;
         }
 
         if(request.getLocale()){
             shopperObject['shopperLocale'] = request.getLocale();
         }
 
-        if(params.Args && params.Args.adyenForm) {
-            if(params.Args.adyenForm.socialSecurityNumber.value){
-                shopperObject['socialSecurityNumber'] = params.Args.adyenForm.socialSecurityNumber.value;
-            }
-            if(params.Args.adyenForm.dateOfBirth.value){
-                shopperObject['dateOfBirth'] = params.Args.adyenForm.dateOfBirth.value;
-            }
-            if(params.Args.adyenForm.telephoneNumber.value){
-                shopperObject['telephoneNumber'] = params.Args.adyenForm.telephoneNumber.value;
-            }
-            else if(params.Args.Order.defaultShipment.shippingAddress.phone){
-                shopperObject['telephoneNumber']= params.Args.Order.defaultShipment.shippingAddress.phone
-            }
+        return shopperObject;
+    },
+
+    createAddressObjects: function (order, jsonObject) {
+        var shippingAddress = order.defaultShipment.shippingAddress;
+        var houseNumberShip;
+        if (shippingAddress.address2) {
+            houseNumberShip = shippingAddress.address2;
         }
+        else {
+            houseNumberShip = "N/A";
+        }
+        jsonObject["countryCode"] = shippingAddress.countryCode.value.toUpperCase();
 
-        return shopperObject;
+        jsonObject.deliveryAddress = {
```

Comment:
Improvement here can be that we set the defaults first and then override with the actual value see: https://github.com/Adyen/adyen-magento2/blob/develop/Helper/Requests.php#L157-L163

Reply:
Done